README.md
![](https://dl.dropbox.com/u/34938401/TT%20Logo2.png)

Welcome to the Thaumic Tinkerer Github Repo.

Thaumic Tinkerer is a Thaumcraft addon, it's intention is to add new content to Thaumcraft, and bring back some oldies, but goodies.

[Thaumic Tinkerer MCF Thread](http://www.minecraftforum.net/topic/1813058-)

Thaumcraft © Azanor 2012
[Thaumcraft MCF Thread](http://www.minecraftforum.net/topic/1585216-)

This mod is made with a Deobfuscated version of Thaumcraft, loaded as a Library. You can get a deobfuscated version of Thaumcraft using immibis' [Bearded Octo Nemesis](https://github.com/immibis/bearded-octo-nemesis).  
![](http://i.creativecommons.org/l/by-nc-sa/3.0/88x31.png)

Thaumic Tinkerer Was originally by Vazkii and is licensed under a [Creative Commons Attribution-NonCommercial-ShareAlike 3.0 Unported License](http://creativecommons.org/licenses/by-nc-sa/3.0/deed.en_GB)


web/changelog
**Build 146**
* Re-added prefix to tile entity names

**Build 145**
* Fixed crash on servers
* Updated German translation
* Changed Ichor textures
* Removed 'ttinkerer:' prefix. This fixes console spam on startup.

**Build 144**
* Updated to 1.7!
* Added new enchantments
* Added mob transmutation

**Build 141**
* Updated to CC1.6 **Warning, this will not work with Computer Craft 1.5**

**Build 140**
* Added SkyMatrix to blacklist, another broken research to avoid.

**Build 139**
* Added ability to set OpenPeripheral mode in configs
* Made funnels respect jars labels
* Removed custom bedrock 
* Added ability for other mods to remove research from KAMI
* Added Aliquo to research blacklist


**Build 138**
* Made OpenPeripheral compatibility changes **Warning: Jar and Other Aspect containers computer craft API has changed, please check wiki**
* Fixed Bedrock Dimension being inaccesible when Galacticraft is installed.
* Fixed WAILA bug showing Warp gates incomming teleports inverted
* Made Kami default on **Warning: config change for kami has changed, to allow for the change**
*Baring any major bugs, this will be the last 1.6.4 release*

**Build 137**
* Fixed Kami requiring Researches AT has not implemented.

**Build 136**
* Added computercraft support for basic essentia pipes
* Fixed some problems with the dislocator
* Fixed problem with tablet and mods that alter Player inventory
* Fixed worldshapers looking glass rendering
* Fixed Relay needing Motus
* Fixed overlapping research
* Made Kami armor toggleable individually, and attached to a keybind
* Fixed various rendering issues

**Build 130**
* Fixed crash cased by kami and mirrors interaction

**Build 127**
* Made erroring focii only available when mirrors enabled
* Made Thaumic Tinkerer add CC API to Void jars
* Attempt to fix crash with updating ichor wands

**Build 101**
* Removed Locomotive's entity movement
* Added blacklist for spawning ores in the bedrock dimension
* Fixed bug where conflicting enchantments could be applied in the enchanter
* Fix error with disabling dimension
* KAMI research is no longer hidden.

**Build 99**
* Update German Translations
* Config file for bedrock dimension ore density
* Updated vis discounting to work with 4.1
* Removed ability of staffs to work in enchanter, at Azanor's request
* Fix withers breaking bedrock
* Fix potential compatibility issues
* Fix AoE pickaxe if bedrock dimension is disabled

**Build 98**
* Fixed Ichorium wand components.

**Build 97**
* Bedrock dimension has 1/6 of the ores. Now it looks sad and lonely.

**Build 96**
* Make some research not-hidden
* Fixed: Excavation focus can break bedrock

**Build 95**
* Changed default dimension ID from 19 to -19

**Build 94**
* Removed special bedrock biome

**Build 93**
* Recreated Thaumic Tinkerer research tree

**Build 92**
* Added config for dimension Id.

**Build 91**
* Added Golem Connector
* Added Tinkers Construct compatibility
* Added Ender storage compatibility
* Altered Dislocation Focus to store block on focus

**Build 75 BETA unstable** 05/02/2014
* Added Levitational Locomotive.
* Fixed issue #189
* Fixed issue #169

**Build 74**
* Fixed server crash

**Build 73**
* Fixed artifice tab Crash

**Build 72**
* WAILA support
* Added Dissipating turtle
* (Flaxbeard) Made sure API for osmotic enchanter was in again.

**Build 68**
* (Flaxbeard) Added an API for the osmotic enchanter.

**Build 67**
* Fixed the Celestial Recall focus not working.
* Added a config option to remove Ore Dictionary recipes, they cause issues if you have Magical Crops and no other mods that add copper.

**Build 66**
* Fixed a DC when using a Shadowbeam focus in a server.
* Fixed a crash when right clicking a XP Drain focus.
* Fixed a dupe involving Dynamism Tablets, bottles and cauldrons.

**Build 65**
* Fixed some foci not being able to be put in wands.
* Changed Awakened Ichorium Tools to change mode with **shift**-right click.

**Build 64**
* Fixed a crash when using Camo blocks to camo Camo blocks (yo dawg).
* Fixed a crash with Easymode research and Lost Research Note.
* Fixed the Leggings of the Burning Mantle not showing the discount tooltip.
* Fixed Nether/Ender shards being extremely common.
* Fixed Hyperenergetic Nitor and Protoclay not being in the creative tab.
* Fixed being able to apply Slow Fall to the Boots of the Horizontal Shield.
* Fixed the flight status not resetting after taking off the Robes of the Stratosphere.
* Fixed a multiplayer bug where the Celestial Network would occasionally open it's GUI for a player who wasn't standing on it.
* Made the Celestial Network GUI easier to navigate.
* Tweaked the code for the XP Drain focus a bit.
* Moved the Shadowbeam Focus research to not be in the same place as a research from Thaumic Exploration.
* Changed the Celestial Pearl recipe to output 2 rather than 1.

**Build 63**
* Fixed Protoclay not having a config option for ID.
* Fixed the XP Drain focus sometimes going over the max value on the wand.

**Build 62**
* Fixed Server Crash.

**Build 61**
* Allow for blocks with directions (i.e. Furnaces, Pumpkins) to keep the direction in camo blocks
* Allow for more blocks to be used with camo blocks (i.e. Wood Logs, Quartz Blocks)
* Fixed the Ethereal Platform breaking instantly.
* Fixed camo blocks not respecting biome colors
* Some typo fixes
* Added a way to turn off the light feature in the Leggings of the Burning Mantle.
* Added a config option to specify dimension IDs (for people who change them)
* (KAMI) Added a few new things ;)

**Build 60**
* Fixed ground level Ethereal Platforms having collision boxes on all sides.
* Fixed the Ichorcloth armor not protecting from damage, it now protects just a little more than diamond.

**Build 59**
(This build is for the TC4.1 beta, do not use it unless you have it).
* Fixed some API/method usage from old versions that doesn't work on 4.1.

**Build 58**
* Updated the ComputerCraft API.

web/index.php
<html>
	<title>Thaumic Tinkerer Files</title>

	<font size="6">Thaumic Tinkerer Files</font><br>
	<a href="http://www.minecraftforum.net/topic/1813058-">Thaumic Tinkerer Thread</a><br>
	<a href="http://vazkii.us">Back to vazkii.us</a>
	<br><hr><br>
	
	<a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/3.0/deed.en_GB"><img alt="Creative Commons Licence" style="border-width:0" src="http://i.creativecommons.org/l/by-nc-sa/3.0/88x31.png" /></a><br />Thaumic Tinkerer by Vazkii is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/3.0/deed.en_GB">Creative Commons Attribution-NonCommercial-ShareAlike 3.0 Unported License</a>.<br>
	This webpage and Thaumic Tinkerer itself are open source. Contribute to the project on <a href="https://github.com/Vazkii/ThaumicTinkerer">GitHub</a>!
	<br><hr><br>
	
	<body link="#111111" vlink="#444466" alink="#000099">
	<a href="https://www.sugarsync.com/pf/D9740002_82596175_752964"><b>ThaumicTinkerer KAMI</a></b><br>
	(Notice, KAMI is just a 1KB file that enables the features in TT, you don't have to update this)<br><br>
	<?php

	$file_url = 'https://raw.github.com/pixlepix/ThaumicTinkerer/website/web/urls.txt';
	$file_contents = file_get_contents($file_url);
	$file_contents_array = array_reverse(explode(PHP_EOL, $file_contents));

	foreach($file_contents_array as $line) {
		$line_props = explode('=', $line);
		$version = $line_props[0];
		$url = $line_props[1];
		
		print("<a href=\"$url\">ThaumicTinkerer <b>$version</b></a><br>");
	}
	?>
	
</body>
</html>